arch/lkl/kernel/time.c

```diff
@@ -121,7 +119,7 @@ void __init time_init(void)
 	if (ret)
 		pr_err("lkl: unable to register clocksource\n");
 
-	clockevents_config_and_register(&clockevent, HZ, 0, 0xffffffff);
+	clockevents_config_and_register(&clockevent, NSEC_PER_SEC, 1, ULONG_MAX);
```

Comment:
I don't think a 0 min_delta is the issue but a combination of min and frequency. When I debugged this, the issue was that even if the delta in clockevents_program_event() was non zero, after applying the min/max/mult/shift transformation the resulting clc was zero.